cli/azd/pkg/tools/azcli/azcli.go

```diff
@@ -1054,34 +983,39 @@ func (cli *azCli) GraphQuery(ctx context.Context, query string, subscriptions []
 		return nil, fmt.Errorf("getting access token: %w", err)
 	}
 
-	client := httputil.GetHttpClient(ctx)
-	headers := map[string]string{
-		"Authorization": fmt.Sprintf("Bearer %s", token.AccessToken),
+	client := http.Client{}
+	requestBuffer := bytes.NewBuffer(requestJson)
+	request, err := http.NewRequest(http.MethodPost, url, requestBuffer)
+	if err != nil {
+		return nil, fmt.Errorf("creating http request: %w", err)
 	}
 
-	request := &httputil.HttpRequestMessage{
-		Url:     url,
-		Method:  "POST",
-		Headers: headers,
-		Body:    string(requestJson),
-	}
+	request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token.AccessToken))
+	request.Header.Set("Content-Type", "application/json")
+	request.Header.Set("Accept", "application/json")
 
-	response, err := client.Send(request)
-	if err != nil || response.Status != http.StatusOK {
+	response, err := client.Do(request)
+	if err != nil || response.StatusCode != http.StatusOK {
 		return nil, fmt.Errorf("sending http request: %w", err)
 	}
 
-	responseText := string(response.Body)
+	defer response.Body.Close()
+	responseBytes, err := io.ReadAll(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("reading http response: %w", err)
+	}
+
+	responseText := string(responseBytes)
 
 	if isNotLoggedInMessage(responseText) {
 		return nil, ErrAzCliNotLoggedIn
 	} else if err != nil {
-		return nil, fmt.Errorf("failed running az graph query: %s: %w", responseText, err)
+		return nil, fmt.Errorf("failed running az graph query: %s: %w", responseBytes, err)
```

Comment:
not sure what the `%s` formatting for `byte[]` will do, but maybe we should use `responseText` here

Reply:
Updated - this block actually isn't required anymore.  Cases for error executing request or reading request body are handled above.  Auth error would be handled in the response status code validation above.